helm-chart/templates/nginx-mesh-api.yaml

```diff
@@ -181,14 +181,84 @@ metadata:
     app.kubernetes.io/part-of: nginx-service-mesh
     spiffe.io/webhook: "true"
 webhooks:
-- name: nginx-mesh-api.sidecar.injector
+  {{- /* Case 1: Namespace is labeled for injection and not in denylist, Pod does not disable injection */ }}
+- name: nginx-mesh-api-ns.injector
   namespaceSelector:
     matchExpressions:
     - key: kubernetes.io/metadata.name
       operator: NotIn
       values:
       - kube-system
       - {{ .Release.Namespace }}
+    matchLabels:
+      injector.nsm.nginx.com/auto-inject: enabled
+  objectSelector:
+    matchExpressions:
```

Comment:
As I think about it, we can probably also filter on the "already injected" label, so that pods that have the sidecar already aren't sent to the controller. Then our controller only needs to check if a pod is the ingress controller to determine if it needs injecting or not.